hcmanager/utils_test.go

```diff
@@ -29,80 +30,44 @@ import (
-	hcManager := NewHardwareClassificationManager(c, klogr.New())
-
-	It("Should fetch BaremetalHosts in ready state and under metal3 namespace", func() {
-		result, _, err := hcManager.FetchBmhHostList(getNamespace())
-		if err != nil {
-			Expect(len(result)).To(BeZero())
-		} else {
-			Expect(len(result)).Should(Equal(2))
-		}
-
-	})
-
-	It("Should return error while fetching BaremetalHosts", func() {
-		_, _, err := hcManager.FetchBmhHostList("sample")
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-
-	})
-
-	It("Should return error for empty hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getEmptyProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should validate extracted hardware profile", func() {
-		result := hcManager.ValidateExtractedHardwareProfile(getExtractedHardwareProfile())
-		if result == nil {
-			Expect(result).Should(BeNil())
-		}
-	})
-
-	It("Should return error for invalid CPU details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidCPUProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for invalid DISK details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidDiskProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for invalid NIC details in hardware profile", func() {
-		hardwareClassification := &hwcc.HardwareClassification{}
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidNicProfile())
-		SetStatus(hardwareClassification, hwcc.ProfileMatchStatusEmpty, hwcc.ProfileMisConfigured, err.Error())
-		if err != nil {
-			Expect(hardwareClassification.Status.ErrorType).Should(Equal(hwcc.ProfileMisConfigured))
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for invalid RAM details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getInvalidRAMProfile())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
-
-	It("Should return error for missing NIC details in hardware profile", func() {
-		err := hcManager.ValidateExtractedHardwareProfile(getMissingNicDetails())
-		if err != nil {
-			Expect(err).To(HaveOccurred())
-		}
-	})
+	DescribeTable("Test fetch BaremetalHost list",
+		func(namespace string) {
+			hostTest := getHosts()
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
+			hcManager := NewHardwareClassificationManager(c, klogr.New())
+			result, _, err := hcManager.FetchBmhHostList(namespace)
+			if err != nil {
+				Expect(len(result)).To(BeZero())
+			}
+			if len(result) > 0 {
+				Expect(len(result)).Should(Equal(2))
+			}
+		},
+		Entry("Should fetch BaremetalHosts in ready state and under metal3 namespace", getNamespace()),
+		Entry("Should return error while fetching BaremetalHosts", "sample"),
+	)
```

Comment:
you should pass what you expect in the function parameter, to verify that it works as you expect it. For example:
```suggestion
	type testCase struct {
		namespace string
		expectError bool
		expectedResult ...
	}
	DescribeTable("Test fetch BaremetalHost list",
		func(tc testCase) {
			hostTest := getHosts()
			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), hostTest...)
			hcManager := NewHardwareClassificationManager(c, klogr.New())
			result, _, err := hcManager.FetchBmhHostList(tc.namespace)
			if tc.ExpectError {
				Expect(err).To(HaveOccured())
				Expect(len(result)).To(BeZero())
			} else {
				Expect(err).NotTo(HaveOccured())
				Expect(len(result)).NotTo(BeZero())
			}
			...
		},
		Entry("Should fetch BaremetalHosts in ready state and under metal3 namespace", testCase{ namespace: getNamespace(), ... }),
		...
	)
```

Reply:
done. @maelk @digambar15 kindly review